File: skills/astra-toolkit/clients/python/examples/tables-create-index-map-keys.py
import os
from astrapy import DataAPIClient

# Get an existing table
client = DataAPIClient()
database = client.get_database(
    os.environ["ASTRA_DB_API_ENDPOINT"], token=os.environ["ASTRA_DB_APPLICATION_TOKEN"]
)
table = database.get_table("**TABLE_NAME**")

# Index a the keys of a map column
table.create_index(
    name="**INDEX_NAME**", column={"**MAP_COLUMN_NAME**": "$keys"}
)

File: skills/astra-toolkit/clients/go/examples/tables-create-index-map-keys.go
package main

import (
	"context"
	"log"
	"os"

	"github.com/datastax/astra-db-go/v2/astra"
	"github.com/datastax/astra-db-go/v2/astra/options"
)

func main() {
	ctx := context.Background()

	// Get an existing table
	client := astra.NewClient()

	database := client.Database(
		os.Getenv("ASTRA_DB_API_ENDPOINT"),
		options.API().SetToken(os.Getenv("ASTRA_DB_APPLICATION_TOKEN")),
	)

	table := database.Table("**TABLE_NAME**")

	// Index a the keys of a map column
	err := table.CreateIndex(
		ctx,
		"**INDEX_NAME**",
		map[string]string{"**MAP_COLUMN_NAME**": "$keys"},
	)
	if err != nil {
		log.Fatal(err)
	}
}
